drivers: phy_ksz8081: Fix latency from autonegotiation

decsny opened the pull request:
Fix ksz8081 driver so that autonegotiation does not block boot.

Fixes #93214

clamattia reviewed (commented):
Boot should not have to wait on `auto-negotiation` see for example: https://github.com/zephyrproject-rtos/zephyr/blob/f52d71c8cf5bfea4e61791cc67ad3f27563f5ba1/drivers/ethernet/phy/phy_mii.c#L312

decsny commented:
it looks like this PR is currently not valid, I was too hasty with it, I made a ticket with problem description to link to it here  https://github.com/zephyrproject-rtos/zephyr/issues/93214

Commit:
phy: ksz8081: Minor refactor of init gpio code

Minor refactor to remove the #ifdefs inline in init function.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
drivers: phy_ksz8081: Fix boot and link up

Two issues are being simultaneously address in this commit:

Issue 1:

Current issue is that zephyr boot / init is slow because it is blocked
by the phy ksz8081 init doing autonegotiation which can take on the
order of a few seconds.

Fix by now doing autonegotiation in the monitor handler instead of in
the cfg link call. The cfg link call will now only set the ANAR register
and a software flag to let the monitor handler know whether or not to
redo the autonegotiation sequence.

Issue 2:

The ksz8081 phy currently does not link up ever on cold boots on the NXP
RT platforms due to regression on mainline. My understanding of why is
not clear, but I found that re-setting the RMII override bit by the time
the monitor work handler runs with the change to fix the first issue,
makes the link come up.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented on drivers/ethernet/phy/phy_microchip_ksz8081.c:
```diff
@@ -329,6 +340,11 @@ static int phy_mc_ksz8081_reset(const struct device *dev)
 	 */
 	k_busy_wait(500 * USEC_PER_MSEC);
 
+	/* After each reset we will apply the static cfg from DT */
+	ret = phy_mc_ksz8081_static_cfg(dev);
+	if (ret) {
+		goto done;
+	}
```
I was following a comment which already existed in the file which said to do this in case of phy reset on every configure, I don't know if the comment is right or not

Commit:
drivers: phy_ksz8081: Improve logging

Improve logging in the driver, using appropriate log levels.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

maass-hamburg commented on drivers/ethernet/phy/phy_microchip_ksz8081.c:
```diff
@@ -427,22 +442,86 @@ static void phy_mc_ksz8081_monitor_work_handler(struct k_work *work)
+	const struct mc_ksz8081_config *config = dev->config;
+
+	if (config->interrupt_gpio.port == NULL) {
+		return -ENODEV;
```
at least have it consistent, because if no instance has that gpio port it will be that
```
#define ksz8081_init_int_gpios(dev) 0
```

Commit:
drivers: ethernet: ksz8081: Save link state for get

Instead of using mdio bus for getting link state, only get it in the
monitor and save it off for get rate api implementation to use

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented:
@maass-hamburg
